error.go

```diff
@@ -0,0 +1,13 @@
+package rabbithole
+
+import "fmt"
+
+type RabbitMQError struct {
```

Comment:
I'd rename this to `ErrorResponse`. There's little RabbitMQ-specific about how RabbitMQ HTTP API reports errors.